app/common/helpers/collections.py

```diff
@@ -323,6 +334,73 @@ def get_previous_question(self, current_question_id: UUID) -> Optional["Question
         raise ValueError(f"Could not find a question with id={current_question_id} in collection={self.collection}")
 
 
+class CollectionHelper:
+    def __init__(self, collection: "Collection", submission_mode: SubmissionModeEnum):
+        self.collection = collection
+        self.submission_mode = submission_mode
+        self.submissions = [
+            s
+            for s in (
+                collection.test_submissions
+                if submission_mode == SubmissionModeEnum.TEST
+                else collection.live_submissions
+            )
+        ]
+        self.submission_helpers = {s.id: SubmissionHelper(s) for s in self.submissions}
+
+    def get_submission_helper_by_id(self, submission_id: UUID) -> SubmissionHelper | None:
+        return self.submission_helpers.get(submission_id, None)
+
+    def get_submission_helper_by_reference(self, submission_reference: UUID) -> SubmissionHelper | None:
+        for _, submission in self.submission_helpers.items():
+            if submission.reference == submission_reference:
+                return submission
+
+        return None
+
+    def get_all_possible_questions_for_collection(self) -> list["Question"]:
+        """
+        Returns a list of all questions that are part of the collection, across all sections and forms.
+        """
+        return [
+            question
+            for section in sorted(self.collection.sections, key=lambda s: s.order)
+            for form in sorted(section.forms, key=lambda f: f.order)
+            for question in sorted(form.questions, key=lambda q: q.order)
+        ]
+
+    def generate_csv_content_for_all_submissions(self) -> str:
+        metadata_headers = ["Submission reference", "Created by", "Created time UTC"]
+        question_headers = {
+            question.id: f"[{question.form.title}] {question.name}"
+            for question in self.get_all_possible_questions_for_collection()
+        }
+        all_headers = metadata_headers + [header_string for _, header_string in question_headers.items()]
+
+        csv_output = StringIO()
+        csv_writer = csv.DictWriter(csv_output, fieldnames=all_headers)
+        csv_writer.writeheader()
+        for submission in [value for key, value in self.submission_helpers.items()]:
+            submission_csv_data = {
+                "Submission reference": submission.reference,
+                "Created by": submission.created_by_email,
+                "Created time UTC": submission.created_at_utc.isoformat(),
+            }
+            visible_questions = submission.all_visible_questions
+            for question_id, header_string in question_headers.items():
+                if question_id not in visible_questions.keys():
+                    submission_csv_data[header_string] = "Not required"
```

Comment:
```suggestion
                    submission_csv_data[header_string] = "NOT_ASKED"
```

Maybe saying 'not asked' instead of 'not required' will distinguish between questions that are mandatory which weren't shown, and questions which are optional and not answered?

Currently all questions are mandatory but not forever, we think.

Possibly also UPPERCASE_UNDERSCORES a bit more indicative of a system value rather than a user entered string?